ios-itunes/ios-itunes/Views/SearchResultsTableViewController.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  SearchResultsTableViewController.swift
+//  ios-itunes
+//
+//  Created by Angel Buenrostro on 1/22/19.
+//  Copyright © 2019 Angel Buenrostro. All rights reserved.
+//
+
+import UIKit
+
+class SearchResultsTableViewController: UITableViewController, UISearchBarDelegate {
+
+    @IBOutlet weak var controlBar: UISegmentedControl!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    let searchResultsController = SearchResultController()
+    
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchTerm = searchBar.text, !searchTerm.isEmpty else { return }
+        
+        var resultType : ResultType!
+        switch controlBar.selectedSegmentIndex {
+        case 0:
+            resultType = .software
+        case 1:
+            resultType = .musicTrack
+        case 2:
+            resultType = .movie
+        default:
+            print("Error setting the appropriate resultType")
+        }
+        
+        searchResultsController.performSearch(searchTerm: searchTerm, resultType: resultType) { (error) in
+            DispatchQueue.main.async {
+//                if error == nil {
+                    self.tableView.reloadData()
+//                }
+        }
+        
+    }
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        searchBar.delegate = self
+        DispatchQueue.main.async {
+            self.searchBarSearchButtonClicked(self.searchBar)
```

Comment:
Actually `searchBar`'s delegate function isn't necessary in here. Because we need an input, a `search term` to perform a search and come back with a data, then reload that data to the table view. But when app just launched we don't need to perform a search
```suggestion

```